WPFSKillTree/ViewModels/PassiveTree/PassiveNodeViewModel.cs

```diff
@@ -0,0 +1,147 @@
+﻿using PoESkillTree.Engine.GameModel;
+using PoESkillTree.Engine.GameModel.PassiveTree;
+using PoESkillTree.Engine.GameModel.PassiveTree.Base;
+using PoESkillTree.Engine.Utils.Extensions;
+using PoESkillTree.SkillTreeFiles;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace PoESkillTree.ViewModels.PassiveTree
+{
+    public class PassiveNodeViewModel
+    {
+        private readonly JsonPassiveNode JsonPassiveNode;
+        public readonly PassiveNodeGroupViewModel? PassiveNodeGroup;
+
+        public PassiveNodeViewModel() : this(new JsonPassiveNode()) { }
+        public PassiveNodeViewModel(JsonPassiveNode jsonPassiveNode, PassiveNodeGroupViewModel? group = null)
+        {
+            JsonPassiveNode = jsonPassiveNode;
+            PassiveNodeGroup = group;
+            InitializeAttributes();
+        }
+        public PassiveNodeDefinition PassiveNodeDefinition => PassiveNodeDefinition.Convert(JsonPassiveNode);
+        public CharacterClass? StartingCharacterClass { get => JsonPassiveNode.StartingCharacterClass; set => JsonPassiveNode.StartingCharacterClass = value; }
```

Comment:
Of all these setters, I found two that are used: StatDescriptions and Id. StatDescriptions can be made private and Id is only used in tests, which are also the only uses of the parameterless test, so it can be a parameter there.

All other setters can be removed. Those properties aren't really something that should be changed anytime.

PassiveNodeGroupViewModel and PassiveTreeViewModel also have a few unused setters.

Reply:
These have been cleaned up